legend-engine-xts-authentication/legend-engine-xt-authentication-connection-factory/src/main/java/org/finos/legend/connection/protocol/AuthenticationMechanism.java

```diff
@@ -22,6 +22,7 @@ public interface AuthenticationMechanism
 
     Class<? extends AuthenticationConfiguration> getAuthenticationConfigurationType();
 
+    // phase 0 - what is this ? delete if not used. We can always add later 
     default AuthenticationConfiguration generateConfiguration()
```

Comment:
It's used to configure which mechanism comes with a config which can be constructed automatically without any inputs so as to support the case where people don't specify which mechanism they would like to use to connect. I'll leave it (potentially clean it up to make it less verbose) and document it.